packages/loader/container-loader/src/containerStorageAdapter.ts

```diff
@@ -73,3 +75,71 @@ export class ContainerStorageAdapter implements IDocumentStorageService {
         return this.storageGetter().createBlob(file);
     }
 }
+
+/**
+ * Storage which only supports createBlob() and readBlob(). This is used with IDetachedBlobStorage to support
+ * blobs in detached containers.
+ */
+export class BlobOnlyStorage implements IDocumentStorageService {
+    constructor(
+        private readonly blobStorage: IDetachedBlobStorage,
+        private readonly logger: ITelemetryLogger,
+    ) { }
+
+    public async createBlob(content: ArrayBufferLike): Promise<ICreateBlobResponse> {
+        return this.blobStorage.createBlob(content);
+    }
+
+    public async readBlob(blobId: string): Promise<ArrayBufferLike> {
+        return this.blobStorage.readBlob(blobId);
+    }
+
+    public get policies(): IDocumentStorageServicePolicies | undefined {
+        this.logger.sendErrorEvent({
+            eventName: "NoRealStorageInDetachedContainer",
+        });
+        throw new Error("Real storage calls not allowed in Unattached container");
+    }
+
+    public get repositoryUrl(): string {
+        this.logger.sendErrorEvent({
+            eventName: "NoRealStorageInDetachedContainer",
+        });
+        throw new Error("Real storage calls not allowed in Unattached container");
+    }
+
+    public async getSnapshotTree(version?: IVersion): Promise<ISnapshotTree | null> {
+        this.logger.sendErrorEvent({
+            eventName: "NoRealStorageInDetachedContainer",
+        });
+        throw new Error("Real storage calls not allowed in Unattached container");
+    }
+
+    public async getVersions(versionId: string, count: number): Promise<IVersion[]> {
+        this.logger.sendErrorEvent({
+            eventName: "NoRealStorageInDetachedContainer",
+        });
+        throw new Error("Real storage calls not allowed in Unattached container");
+    }
+
+    public async write(tree: ITree, parents: string[], message: string, ref: string): Promise<IVersion> {
+        this.logger.sendErrorEvent({
+            eventName: "NoRealStorageInDetachedContainer",
+        });
+        throw new Error("Real storage calls not allowed in Unattached container");
+    }
+
+    public async uploadSummaryWithContext(summary: ISummaryTree, context: ISummaryContext): Promise<string> {
+        this.logger.sendErrorEvent({
+            eventName: "NoRealStorageInDetachedContainer",
+        });
+        throw new Error("Real storage calls not allowed in Unattached container");
+    }
+
+    public async downloadSummary(handle: ISummaryHandle): Promise<ISummaryTree> {
```

Comment:
I wonder if we want to reduce a bit impact on bundle size (if construct below results in less code) by only logging stack (as part of our telemetry reporting) and not care about logging individual names of methods called. Stack will contain same info (public methods are not renamed), and more (where call is coming from).
Something like this:

```typescript
public downloadSummary: (handle: ISummaryHandle) => Promise<ISummaryTree> = this.notCalled;

private notCalled(): never {
   try {
      throw new Error("BlobOnlyStorage not implemented method used"); // some browsers may not populate stack unless exception is thrown
   } catch (err) {
       this.logger.sendErrorEvent({
               eventName: "BlobOnlyStorageWrongCall",
               err,
           });
       throw err;
   }
}
```

Reply:
Is there an easy way to compare the bundle sizes with/without this change?